src/generic.jl

```diff
@@ -1236,28 +1236,24 @@ function (\)(A::AbstractMatrix, B::AbstractVecOrMat)
     require_one_based_indexing(A, B)
     m, n = size(A)
     if m == n
-        if istril(A)
-            if istriu(A)
-                return Diagonal(A) \ B
-            elseif istriu(A, -1)
+        istrium1 = istriu(A, -1)
+        istril1 = istril(A, 1)
+        if istril1 && iszero(diagview(A,1)) # istril(A)
+            if istrium1
+                if iszero(diagview(A, -1))
+                    return Diagonal(A) \ B
+                end
                 return Bidiagonal(A, :L) \ B
             else
                 return LowerTriangular(A) \ B
             end
         end
-        if istriu(A)
-            if istril(A, 1)
+        if istrium1 && iszero(diagview(A, -1)) # istriu(A)
+            if istril1
                 return Bidiagonal(A, :U) \ B
             end
             return UpperTriangular(A) \ B
         end
-        if isbanded(A, -1, 1)
```

Comment:
Wouldn't you get this "for free" with `istrium1 && istril1`?

Reply:
Yes, but there were some triangular `lu`-related test failures for unusual types, so I've removed the triangular branch for now. We may add this back once those issues are resolved.